manifests/0000_90_kube-apiserver-operator_04_servicemonitor-apiserver.yaml

```diff
@@ -138,3 +138,121 @@ spec:
     - record: cluster:apiserver_current_inflight_requests:sum:max_over_time:2m
       expr: |
         max_over_time(sum(apiserver_current_inflight_requests{apiserver=~"openshift-apiserver|kube-apiserver"}) by (apiserver,request_kind)[2m:])
+  - name: api-performance
+    rules:
+    - record: verb_resource:apiserver_request_duration_seconds_bucket:rate:1m
+      expr: sum(rate(apiserver_request_duration_seconds_bucket{apiserver=~"openshift-apiserver|kube-apiserver",subresource!="log",verb!~"WATCH|WATCHLIST|PROXY"}[1m])) by (apiserver, resource, verb, le)
```

Comment:
notes for the future:
```
apiserver=~"openshift-apiserver|kube-apiserver"
```
we should also most likely include the oauth-apiserver to this dashboard.

```
subresource!="log"
```
this needs to be updated to `subresource!~"proxy|attach|log|exec|portforward"` based on what we recently changed in the SLO alerts

we might also want to use `apiserver_request_sli_duration_seconds_bucket` that excludes pnf & webhooks latency  instead of the e2e latency metric

Reply:
>we should also most likely include the oauth-apiserver to this dashboard.

Neither of the mentioned metrics has oauth-apiserver in `apiserver` label

Reply:
>>subresource!="log"

>this needs to be updated to subresource!~"proxy|attach|log|exec|portforward" based on what we recently changed in the SLO alerts

good idea, done

Reply:
>we might also want to use apiserver_request_sli_duration_seconds_bucket that excludes pnf & webhooks latency instead of the e2e latency metric

I think its best being done in the new PR - to make this PR just a pure recording rules transition - also not sure how far can we backport this

Reply:
> Neither of the mentioned metrics has oauth-apiserver in apiserver label

maybe it has another value, but there should be metrics for the oauth-apiserver deployed by https://github.com/openshift/cluster-authentication-operator/ and we most likely also want to incude them in the dashboards in the future

Reply:
> I think its best being done in the new PR - to make this PR just a pure recording rules transition - also not sure how far can we backport this

yeah I totally agree. My intention with the original comment was to take notes of things I noticed while reviewing to fix later

Reply:
Looking at an OCP cluster, we indeed don't have any metrics for the oauth-apiserver, that's pretty odd, maybe we are just not exposing any metrics there

Reply:
So we are not collection metrics for the openshift-apiserver at all despite the fact that it has a ServiceMonitor, but it is not picked up by Prometheus. I opened https://github.com/openshift/cluster-authentication-operator/pull/616 in an attempt to fix that.